menu.js

```diff
@@ -1,5 +1,48 @@
 const menu = {
-  // TODO
+  //task 1
+  _courses: {
+    appetizers: [],
+    mains: [],
+    desserts: []
+  },
+
+  get courses() {
+    return this._courses;
+  },
+
+  addDishToCourse(courseName, dishName, dishPrice) {
+    const dish = {
+      name: dishName,
+      price: dishPrice
+    };
+    if (courseName === 'appetizers') {
+      menu._courses.appetizers.push(dish);
+    } else if (courseName === 'mains') {
+      menu._courses.mains.push(dish);
+    } else if (courseName === 'desserts') {
+      menu._courses.desserts.push(dish);
+    }
+  },
+
+  getRandomDishFromCourse(courseName) {
+    let mealIndex = Math.floor(Math.random() * this._courses[courseName].length);
+    return this._courses[courseName][mealIndex];
```

Comment:
You are calling `this._courses[courseName]` 2 times. I think making extra variable for its value is a good idea. Something like `courseDishes` should be fine.

Reply:
At first it was as you suggested...